src/STAT_FrontEnd.C

```diff
@@ -1098,7 +755,7 @@ StatError_t STAT_FrontEnd::connectMrnetTree(bool blocking)
     if (blocking == false)
     {
         sConnectAttempt = sConnectAttempt + 1;
-        if (!WIFBESPAWNED(gsLmonState))
+        if (daemonsHaveExited())
```

Comment:
Pulled this check into a virtual function, since it tends to show up a lot in code that is otherwise launcher agnostic.

bool STAT_lmonFrontEnd::daemonsHaveExited()
{
    return !WIFBESPAWNED(gsLmonState);
}